jupyterhub/templates/hub/deployment.yaml

```diff
@@ -70,6 +70,15 @@ spec:
             secretKeyRef:
               name: hub-secret
               key: proxy.token
+        {{ range $key, $value := .Values.hub.services -}}
```

Comment:
A more robust way to do this would be to mount the secret as a volume (similar to how we do for configmap) and just read it with something similar to get_config we currently use in jupyterhub_config.py. This will break with strange errors if there are non alphanumeric characters in service name, for example.

Reply:
Do you mean put the whole `hub.services` in a secret volume instead of splitting like this?

Reply:
No, we could mount hub-secret into a volume, similar to us mounting the configmap. So all secrets would be in /etc/jupyterhub/secrets are somesuch, and we can have a `get_secret` method in jupyterhub_config.py similar to `get_config`. 

I'm also happy to make that change as a separate PR after this.

Reply:
Roger. I think that's enough for me to give it a go.

Reply:
OK, I think I got it. I added a `src` for `get_config` to fetch from either `config` or `secret` and put the whole services dict in the secret since it contains secrets and is much simpler than splitting it between configmap and secrets. Let me know if this implementation makes sense. I tested it and it does work, at least.